### PR TITLE
update documentation to make optional things use optional brackets

### LINKED ISSUE #7106: Certain parameters in API reference are inconsistent in marking optional parameters
**Describe the bug (REQUIRED)**
Select optional parameters are inconsistent in how they are marked in the API reference.  Namely go.get, go.set, and sys.sys_getsysinfo.  They do not flag optional parameters as [paramname], and instead merely mark it as optional in the description for the parameter

**To Reproduce (REQUIRED)**
Observe documentation.
These can be seen here:
https://defold.com/ref/stable/go/#go.get:url-property-options
https://defold.com/ref/stable/go/#go.set:url-property-value-options
https://defold.com/ref/stable/sys/#sys.get_sys_info:options

Compare this to:
https://defold.com/ref/stable/go/#go.get_position:[id]

and observe that the optional parameter is marked by the parameter name being marked in `[]`.

This markup is helpful when using automated tools to generate language bindings from the API.

**Expected behavior (REQUIRED)**
* Optional parameters are marked by square brackets.

**Defold version (REQUIRED):**
  - 1.3.7

**Platforms (REQUIRED):**
* All

### PR DESCRIPTION
**Pull request by RealityStop**
update documentation to make optional things use optional brackets notation

Fix #7106

**Please note that I do not have the toolchain set up currently to build the project and ensure there are no side-effects to these changes.   The changes should only affect the API reference, but I have merely copied the style of the existing documentation.  I can't test these changes myself yet.**

### CODE REVIEW

**Comment by RealityStop** [comment]
Hmmm... well, I HAVE signed the CLA, but it's being stubborn.  I'mma tryin'

File: engine/gameobject/src/gameobject/gameobject_script.cpp [hunk 1]
```diff
@@ -550,7 +550,7 @@ namespace dmGameObject
      * @name go.get
      * @param url [type:string|hash|url] url of the game object or component having the property
      * @param property [type:string|hash] id of the property to retrieve
-     * @param options [type:table] (optional) options table
+     * @param [options] [type:table] optional options table
```
Comment (JCash) [inline engine/gameobject/src/gameobject/gameobject_script.cpp:553]: Question is, would this actually work with our current scripts?
These changes aren't tested, right?

Comment (britzl) [reply engine/gameobject/src/gameobject/gameobject_script.cpp:553]: Let me test this!

Comment (britzl) [reply engine/gameobject/src/gameobject/gameobject_script.cpp:553]: Generated to this in `go_doc.script_api`:

```
  - desc: gets a named property of the specified game object or component
    name: get
    parameters:
    - desc: url of the game object or component having the property
      name: url
      type:
      - string
      - hash
      - url
    - desc: id of the property to retrieve
      name: property
      type:
      - string
      - hash
    - desc: 'optional options table

        - index [type:integer] index into array property (1 based)

        - key [type:hash] name of internal property'
      name: options
      optional: true
      type: table
    returns:
    - desc: the value of the specified property
      type: any
    type: function
```

BUT the `go_doc.json` doesn't look quite right:


```
    {
      "type": "FUNCTION",
      "name": "go.get",
      "brief": "gets a named property of the specified game object or component",
      "description": "gets a named property of the specified game object or component",
      "returnvalues": [
        {
          "name": "value",
          "doc": "the value of the specified property",
          "types": [
            "any"
          ]
        }
      ],
      "parameters": [
        {
          "name": "url",
          "doc": "url of the game object or component having the property",
          "types": [
            "string",
            "hash",
            "url"
          ]
        },
        {
          "name": "property",
          "doc": "id of the property to retrieve",
          "types": [
            "string",
            "hash"
          ]
        },
        {
          "name": "[options]",
          "doc": "optional options table\n- index <span class=\"type\">integer</span> index into array property (1 based)\n- key <span class=\"type\">hash</span> name of internal property",
          "types": [
            "table"
          ]
        }
      ],
      "examples": "Get a named property\n<div class=\"codehilite\"><pre><span></span><code><span class=\"kr\">function</span> <span class=\"nf\">init</span><span class=\"p\">(</span><span class=\"n\">self</span><span class=\"p\">)</span>\n    <span class=\"c1\">-- get the resource of a certain gui font</span>\n    <span class=\"kd\">local</span> <span class=\"n\">font_hash</span> <span class=\"o\">=</span> <span class=\"n\">go</span><span class=\"p\">.</span><span class=\"n\">get</span><span class=\"p\">(</span><span class=\"s2\">&quot;#gui&quot;</span><span class=\"p\">,</span> <span class=\"s2\">&quot;fonts&quot;</span><span class=\"p\">,</span> <span class=\"p\">{</span><span class=\"n\">key</span> <span class=\"o\">=</span> <span class=\"s2\">&quot;system_font_BIG&quot;</span><span class=\"p\">})</span>\n<span class=\"kr\">end</span>\n</code></pre></div>",
      "replaces": "",
      "error": "",
      "tparams": [],
      "members": [],
      "notes": []
    },
```

I was expecting the `options` table to be set as optional, like in the script_api. Perhaps this was not how it worked before either?

Comment (britzl) [reply engine/gameobject/src/gameobject/gameobject_script.cpp:553]: > Perhaps this was not how it worked before either?

Correct, that part wasn't working before either.

Comment (britzl) [reply engine/gameobject/src/gameobject/gameobject_script.cpp:553]: We are checking for optional arguments in the generator:

https://github.com/defold/defold/blob/dev/engine/docs/src/script_doc.py#L330-L335

**Review by britzl — APPROVED** [review]

**Pull request merged**